File: Remove_Duplicates_from_Sorted_Array_II.java
// TC : O(n)
// SC : O(1)
class Solution {
    public int removeDuplicates(int[] nums) {
        
        if(nums == null || nums.length == 0) return 0;
                
        int j = 0;
        
        for(int k : nums)
            if(j < 2 || nums[j-2] != k)
                nums[j++] = k;
        
        return j;
    }
}

File: remove_duplicates_from_sorted_array_ii.py
# Time Complexity: O(n) where n is the length of the array 
# Space Complexity:  O(1)
#  Did this code successfully run on Leetcode : Yes
#  Any problem you faced while coding this : No

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 1 # to iterate over the array
        j = 1 # to maintain index of result array.
        count = 1
        while(i < len(nums)):
            if nums[i] == nums[i-1]:
                count += 1
            else:
                count = 1
            if count <= 2:
                nums[j] = nums[i]
                j+= 1
            i += 1
        return j
